entity-framework/core/miscellaneous/nativeaot-and-precompiled-queries.md

```diff
@@ -0,0 +1,110 @@
+---
+title: NativeAOT Support and Precompiled Queries (Experimental) - EF Core
+description: Publishing NativeAOT Entity Framework Core applications and using precompiled queries
+author: roji
+ms.date: 11/10/2024
+uid: core/miscellaneous/nativeaot-and-precompiled-queries
+---
+# NativeAOT Support and Precompiled Queries (Experimental)
+
+> [!WARNING]
+> NativeAOT and query precompilation are highly experimental feature, and are not yet suited for production use. The support described below should be viewed as infrastructure towards the final feature, which will likely be released with EF 10. We encourage you to experiment with the current support and report on your experiences, but recommend against deploying EF NativeAOT applications in production. See below for specific known limitations.
+
+[.NET NativeAOT](/dotnet/core/deploying/native-aot) allows publishing self-contained .NET applications that have been compiled ahead-of-time (AOT). Doing so offers the following advantages:
+
+* Significantly faster application startup time
+* Small, self-contained binaries that have smaller memory footprints and are easier to deploy
```

Comment:
What's FDD? :) Are you proposing to change "small" to "smaller"?

Reply:
FDD = [Framework Dependent Deployment](https://learn.microsoft.com/en-us/dotnet/core/deploying/deploy-with-cli#framework-dependent-deployment).

I would, personally and generally, avoid talking about binary size here. :)

Reply:
Comparing to FDD is a bit disingenuous, since you have to have the full .NET runtime installed, no? A hello world NativeAOT app is a few MBs, whereas the same thing with FDD is far bigger, if you count the runtime...

In any case, regardless of EF, smaller binary size is one of the main points of NativeAOT overall - it's certainly  mentioned all around...